pkg/reconciler/events/storage/storage.go

```diff
@@ -76,117 +75,29 @@ type Reconciler struct {
-	} else if updated {
-		// There was a difference and updateFinalizers said it updated and did not return an error.
-		r.Recorder.Eventf(storage, corev1.EventTypeNormal, "Updated", "Updated CloudStorageSource %q finalizers", storage.Name)
-	}
-
-	if equality.Semantic.DeepEqual(original.Status, storage.Status) {
-		// If we didn't change anything then don't call updateStatus.
-		// This is important because the copy we loaded from the informer's
-		// cache may be stale and we don't want to overwrite a prior update
-		// to status with this stale state.
-
-	} else if uErr := r.updateStatus(ctx, original, storage); uErr != nil {
-		logging.FromContext(ctx).Desugar().Warn("Failed to update CloudStorageSource status", zap.Error(uErr))
-		r.Recorder.Eventf(storage, corev1.EventTypeWarning, "UpdateFailed",
-			"Failed to update status for CloudStorageSource %q: %v", storage.Name, uErr)
-		return uErr
-	} else if reconcileErr == nil {
-		// There was a difference and updateStatus did not return an error.
-		r.Recorder.Eventf(storage, corev1.EventTypeNormal, "Updated", "Updated CloudStorageSource %q", storage.Name)
-	}
-	if reconcileErr != nil {
-		r.Recorder.Event(storage, corev1.EventTypeWarning, "InternalError", reconcileErr.Error())
-	}
-	return reconcileErr
-}
-
-func (r *Reconciler) reconcile(ctx context.Context, storage *v1alpha1.CloudStorageSource) error {
+func (r *Reconciler) ReconcileKind(ctx context.Context, storage *v1alpha1.CloudStorageSource) reconciler.Event {
 	ctx = logging.WithLogger(ctx, r.Logger.With(zap.Any("storage", storage)))
 
 	storage.Status.InitializeConditions()
-
-	// See if the source has been deleted.
-	if storage.DeletionTimestamp != nil {
-		logging.FromContext(ctx).Desugar().Debug("Deleting CloudStorageSource notification")
-		if err := r.deleteNotification(ctx, storage); err != nil {
-			storage.Status.MarkNotificationNotReady("NotificationDeleteFailed", "Failed to delete CloudStorageSource notification: %s", err.Error())
-			return err
-		}
-		storage.Status.MarkNotificationNotReady("NotificationDeleted", "Successfully deleted CloudStorageSource notification: %s", storage.Status.NotificationID)
-
-		if err := r.PubSubBase.DeletePubSub(ctx, storage); err != nil {
-			return err
-		}
-
-		// Only set the notificationID to empty after we successfully deleted the PubSub resources.
-		// Otherwise, we may leak them.
-		storage.Status.NotificationID = ""
-		removeFinalizer(storage)
-		return nil
-	}
-
-	// Ensure that there's finalizer there, since we're about to attempt to
-	// change external state with the topic, so we need to clean it up.
-	addFinalizer(storage)
+	storage.Status.ObservedGeneration = storage.Generation
 
 	topic := resources.GenerateTopicName(storage)
 	_, _, err := r.PubSubBase.ReconcilePubSub(ctx, storage, topic, resourceGroup)
 	if err != nil {
-		return err
+		return reconciler.NewEvent(corev1.EventTypeWarning, "ReconcilePubSubFailed", "Failed to reconcile CloudStorageSource PubSub: %s", err)
```

Comment:
nit to maintain consistency with others:
ReconcilePubSubFailed -> PubSubReconcileFailed

Reply:
done